File: telegram/src/core/entities/Notification.ts
export class Notification {
    type!: "email" | "telegram" | "push";
    address: string;
    title: string;
    message: string;
    status?: string;
    retryCount?: number;


    constructor(address: string, title: string, message: string, status?: string, retryCount: number = 0, type: "email" | "telegram" | "push" = "telegram") {
        this.type = type;
        this.address = address;
        this.title = title;
        this.message = message;
        this.status = status;
        this.retryCount = retryCount;
    }
}

File: push/app/domain/notification.py
class Notification:
    def __init__(self, notification_type: str, address: str, title: str, message: str):
        # self.id = id
        self.notification_type = notification_type
        self.address = address
        self.title = title
        self.message = message

    def __repr__(self) -> str:
        return f"<Notification {self.id}>"
